Client Protocol Mappers with non UUID ids cannot be edited

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

In [MappingDetails.tsx](https://github.com/keycloak/keycloak/blob/440624e39821689fa9364c6552a7ac2b8b604716/js/apps/admin-ui/src/client-scopes/details/MappingDetails.tsx#L49-L50) a check is made if the passed route parameter `mapperId` is a GUID or not.

If it is a GUID, we assume an existing Mapper is updated.
If it is not a GUID, we assume a new Mapper will be added.

Unfortunately, nothing stops you from passing a non-UUID as an id into the createMapper Request in `ProtocolMappersResource` or via SQL script. (we did this when we automated the creation of the "same" client in different realms)

The result is that it is not possible to edit those Mappers because the UI thinks it is a new one but then does not find it and throws an error.

```
react-dom.production.min.js:10 Error: Could not find the resource that you are looking for
    at We.z.config (MappingDetails.tsx:110:17)
```



### Version

26.0.5

### Regression

- [X] The issue is a regression

### Expected behavior

Protocol Mapper without a GUID as ID can be edited.

### Actual behavior

Protocol Mapper without a GUID as ID can not be edited.

### How to Reproduce?

1) Add a Protocol Mapper via REST endpoint that has not a valid GUID as id 
```
POST https://kc-host/auth/admin/realms/yyy/clients/2815b512-f45a-4e03-826a-23f6c8845a6b/protocol-mappers/models
Authorization: Bearer {{auth_token}}
Content-Type: application/json

{
  "id": "not-a-guid",
  "protocol": "openid-connect",
  "protocolMapper": "oidc-audience-mapper",
  "name": "test-mapper",
  "config": {
    "included.client.audience": "account",
    "included.custom.audience": "",
    "id.token.claim": "false",
    "access.token.claim": "true",
    "introspection.token.claim": "true"
  }
}
```

2) Try to edit the Protocol Mapper in the UI

### Anything else?

_No response_

--- a/js/apps/admin-ui/src/client-scopes/EditClientScope.tsx
+++ b/js/apps/admin-ui/src/client-scopes/EditClientScope.tsx
@@ -185,6 +185,7 @@ export default function EditClientScope() {
           realm,
           id: clientScope!.id!,
           mapperId: mapper.id!,
+          viewMode: "new",
         }),
       );
     } else {
@@ -256,7 +257,12 @@ export default function EditClientScope() {
               onAdd={addMappers}
               onDelete={onDelete}
               detailLink={(id) =>
-                toMapper({ realm, id: clientScope.id!, mapperId: id! })
+                toMapper({
+                  realm,
+                  id: clientScope.id!,
+                  mapperId: id!,
+                  viewMode: "edit",
+                })
               }
             />
           </Tab>

--- a/js/apps/admin-ui/src/client-scopes/details/MappingDetails.tsx
+++ b/js/apps/admin-ui/src/client-scopes/details/MappingDetails.tsx
@@ -34,7 +34,7 @@ export default function MappingDetails() {
   const { t } = useTranslation();
   const { addAlert, addError } = useAlerts();
 
-  const { id, mapperId } = useParams<MapperParams>();
+  const { id, mapperId, viewMode } = useParams<MapperParams>();
   const form = useForm();
   const { setValue, handleSubmit } = form;
   const [mapping, setMapping] = useState<ProtocolMapperTypeRepresentation>();
@@ -46,8 +46,7 @@ export default function MappingDetails() {
   const navigate = useNavigate();
   const { realm } = useRealm();
   const serverInfo = useServerInfo();
-  const isGuid = /^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$/;
-  const isUpdating = !!isGuid.exec(mapperId);
+  const isUpdating = viewMode === "edit";
 
   const isOnClientScope = !!useMatch(MapperRoute.path);
   const toDetails = () =>

--- a/js/apps/admin-ui/src/client-scopes/routes/Mapper.tsx
+++ b/js/apps/admin-ui/src/client-scopes/routes/Mapper.tsx
@@ -7,12 +7,13 @@ export type MapperParams = {
   realm: string;
   id: string;
   mapperId: string;
+  viewMode: "edit" | "new";
 };
 
 const MappingDetails = lazy(() => import("../details/MappingDetails"));
 
 export const MapperRoute: AppRouteObject = {
-  path: "/:realm/client-scopes/:id/mappers/:mapperId",
+  path: "/:realm/client-scopes/:id/mappers/:mapperId/:viewMode",
   element: <MappingDetails />,
   breadcrumb: (t) => t("mappingDetails"),
   handle: {

--- a/js/apps/admin-ui/src/clients/routes/Mapper.tsx
+++ b/js/apps/admin-ui/src/clients/routes/Mapper.tsx
@@ -7,6 +7,7 @@ export type MapperParams = {
   realm: string;
   id: string;
   mapperId: string;
+  viewMode: "edit" | "new";
 };
 
 const MappingDetails = lazy(
@@ -14,7 +15,7 @@ const MappingDetails = lazy(
 );
 
 export const MapperRoute: AppRouteObject = {
-  path: "/:realm/clients/:id/clientScopes/dedicated/mappers/:mapperId",
+  path: "/:realm/clients/:id/clientScopes/dedicated/mappers/:mapperId/:viewMode",
   element: <MappingDetails />,
   breadcrumb: (t) => t("mappingDetails"),
   handle: {

--- a/js/apps/admin-ui/src/clients/scopes/DedicatedScopes.tsx
+++ b/js/apps/admin-ui/src/clients/scopes/DedicatedScopes.tsx
@@ -60,6 +60,7 @@ export default function DedicatedScopes() {
           realm,
           id: client.id!,
           mapperId: mapper.id!,
+          viewMode: "new",
         }),
       );
     } else {
@@ -122,7 +123,7 @@ export default function DedicatedScopes() {
               onAdd={addMappers}
               onDelete={onDeleteMapper}
               detailLink={(mapperId) =>
-                toMapper({ realm, id: client.id!, mapperId })
+                toMapper({ realm, id: client.id!, mapperId, viewMode: "edit" })
               }
             />
           </Tab>

